src/main/java/edu/ksu/canvas/attendance/form/InputValidator.java

```diff
@@ -0,0 +1,48 @@
+package edu.ksu.canvas.attendance.form;
+
+import org.springframework.stereotype.Component;
+import org.springframework.validation.Errors;
+import org.springframework.validation.Validator;
+
+@Component
+public class InputValidator implements Validator {
+
+    @Override
+    public boolean supports(Class<?> clazz) {
+        return CourseConfigurationForm.class.isAssignableFrom(clazz);
+    }
+
+    @Override
+    public void validate(Object target, Errors errors) {
+
+        CourseConfigurationForm classSetupForm = (CourseConfigurationForm) target;
+
+        String assignmentPoints = classSetupForm.getAssignmentPoints();
+        String presentPoints = classSetupForm.getPresentPoints();
+        String tardyPoints = classSetupForm.getTardyPoints();
+        String absentPoints = classSetupForm.getAbsentPoints();
+        String excusedPoints = classSetupForm.getExcusedPoints();
+
+        if (classSetupForm.getGradingOn()){
+            boolean typeValid = (assignmentPoints == null || assignmentPoints.matches("[0-9.]+"))
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of conditional operators (9) used in the expression (maximum allowed 3). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1067)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "[0-9.]+" 10 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1192)